indigo_app/js/components/DocumentTOCView.vue

```diff
@@ -0,0 +1,297 @@
+<template>
+  <t-o-c-controller
+      :items="tocItems"
+      ref="tocController"
+      @on-title-click="onTitleClick"
+  >
+    <template v-slot:search="{ model, clearTitleQuery }">
+      <div class="input-group mb-2">
+        <input type="text"
+               class="form-control form-control-sm"
+               placeholder="Search by title"
+               v-model="model.titleQuery"
+        >
+        <div class="input-group-prepend">
+          <button class="btn btn-sm btn-secondary"
+                  type="button"
+                  @click="clearTitleQuery"
+                  :disabled="!model.titleQuery"
+          >
+            <i class="fas fa-times"></i>
+          </button>
+        </div>
+      </div>
+    </template>
+    <template v-slot:expand-collapse-toggle="{ expandAll, collapseAll }">
+      <div class="d-flex mb-2">
+        <button @click="expandAll" class="btn btn-primary btn-sm mr-1">Expand All</button>
+        <button @click="collapseAll" class="btn btn-primary btn-sm">Collapse All</button>
+      </div>
+    </template>
+  </t-o-c-controller>
+</template>
+
+<script>
+import TOCController from "./toc-controller/index.vue";
+
+export default {
+  name: 'DocumentTOCView',
+  components: {
+    TOCController
+  },
+  props: {
+    backBoneContext: {
```

Comment:
this can fall away and be replaced by the few attributes on backBoneContext that are used in this code.

Reply:
refactored